service/authenticator.go

```diff
@@ -79,10 +79,14 @@ func (c *Config) Authenticate(neg, addr string) (i goidentity.Identity, ok bool,
 		err = fmt.Errorf("SPNEGO negotiation token is not a NegTokenInit: %v", err)
 		return
 	}
-	if !spnego.NegTokenInit.MechTypes[0].Equal(gssapi.MechTypeOIDKRB5) {
-		err = errors.New("SPNEGO OID of MechToken is not of type KRB5")
-		return
-	}
+	krb5Found := false
+        for i := 0 ; i < len(spnego.NegTokenInit.MechTypes) ; i++ {
```

Comment:
Using the range keyword would simplify this section.
Also best to break the loop once the match is found to avoid wasting time further in the loop.
```go
for _, t := range spnego.NegTokenInit.MechTypes {
    if t.Equal(gssapi.MechTypeOIDKRB5) {
        krb5Found = true
        break
    }
}